### PR TITLE
First iteration of script to store data locally.

### PR DESCRIPTION
**Pull request by tommertron**
"covid data getter" is meant to be run on a cron job that perpetually checks against a local CSV for blank values and fills them, primarily for the purpose of accessing the data more easily later. Eventually, there will be a second script that checks the local database and compiles and emails the report to subscribers, and eventually eventually, a way to generate visual charts locally without having to rely on google docs.

### CODE REVIEW

File: covid data getter.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	localData.loc[row,column] = value
+	localData.to_csv(file)
+
+# Function to add a given date as a new row to the CSV 
+def dateCheck(day):
+	try:
+		localData.loc[day]
+		return 'Date found'
+	except KeyError:
+		return 'Date not found'
```
Comment (waffle2k) [inline covid data getter.py:74]: I would suggest returning `None` here

File: covid data getter.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+# Function to add given data to a given row and column 
+def addValue(row,column,value):
+	localData.loc[row,column] = value
+	localData.to_csv(file)
+
+# Function to add a given date as a new row to the CSV 
+def dateCheck(day):
+	try:
+		localData.loc[day]
+		return 'Date found'
```
Comment (waffle2k) [inline covid data getter.py:72]: Just return the `localData.loc[day]` value.  If it throws an exception, then you catch it below and return `None`.  This makes reading this a little easier, as you don't have to know the magic string being returned

File: covid data getter.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	except KeyError:
+		return 'Date not found'
+
+# Function to check blank fields for a given date 
+def blankchecker(date,dataset):
+	fieldquery = []
+	for i in datasources[dataset]['fields']:
+		try:
+			checker = 1 + int(localData.loc[date,i])
```
Comment (waffle2k) [inline covid data getter.py:81]: Is `checker` ever used?  If there's nothing being evaluated?

File: covid data getter.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	urlstart = 'https://data.ontario.ca/api/3/action/datastore_search?'
+	resourceid = 'resource_id=' + datasources[dataset]['id']
+	fieldQuery = 'fields='
+	numfields = len(qfields)
+	for x in qfields:
+		comma = ','
+		if numfields == 1:
+			comma = ''
+		fieldQuery = fieldQuery+'\"'+x+'\"' + comma
+		numfields -= 1
```
Comment (waffle2k) [inline covid data getter.py]: Just replace this with:

`fieldQuery = "fields=%s" % (",".join(qfields))`

**Review by waffle2k — COMMENTED** [review]
Just commenting on some style portions, without looking at the CSV part of it.  So far it looks really good, and I don't want to suggest too many changes to a working solution, but rather suggest potential iterative improvements: Once you get your head around classes, refactor the CSV related stuff into a class so that you're not dealing with global variables and such.  It's out of scope for this change, but I think that it'll make reading this a lot easier (and also working on this easier) if you're not having to remember all of the moving parts.

Looks great otherwise!!

File: covid data getter.py (hunk 3 above)
Comment (tommertron) [reply covid data getter.py:81]: This is most probably not the best way to do this, but it's a way of checking whether there are blanks in a record's field. For some reason I couldn't get it to work by checking for 'None'.  The pandas library doesn't seem to see blanks as 'None'. So I just force a type error by doing some math to tell me if it's blank. 🤷‍♂️

File: covid data getter.py (hunk 2 above)
Comment (tommertron) [reply covid data getter.py:72]: That makes a lot of sense. Something wasn't sitting right with me with the way I did this.

File: covid data getter.py (hunk 3 above)
Comment (waffle2k) [reply covid data getter.py:81]: Yeah, ok I get it. That's super annoying

**Pull request merged**